sdk/worker/task_runner.go

```diff
@@ -469,6 +479,7 @@ func (c *TaskRunner) increaseRunningWorkers(taskName string) error {
 	c.runningWorkersByTaskNameMutex.Lock()
 	defer c.runningWorkersByTaskNameMutex.Unlock()
 	c.runningWorkersByTaskName[taskName] += 1
+	c.workerWaitGroup.Add(1)
```

Comment:
A running worker also increments the workerWaitGroup counter because we want `WaitWorkers()` to wait for actively running workers.